### PR TITLE
Add possible errors info to JSON RPC methods

### PR DESCRIPTION
**Pull request by khorolets**
Introduce as much as possible information about possible errors for each JSON RPC method we have. Yet there is a lot of copy-paste data I believe it's one of the most readable ways to show errors.

This PR needs some attention to the language because it's me, you know.

**NB** this PR is based on #801 I am marking this as a draft until that one is merged.

### CODE REVIEW

**Comment by bucanero** [comment]
Hi @khorolets,
#801 has been merged, so you should be unblocked now 👍

**Comment by khorolets** [comment]
> Hi @khorolets,
> #801 has been merged, so you should be unblocked now 👍

Thanks for the heads up! I've resolved conflicts and marked PR as ready for review!

**Review by frol — CHANGES_REQUESTED** [review]
> The requested block can't be found in a database

I suggest considering the following: "The requested block has not been produced yet or it has been garbage-collected (cleaned up to save space on the RPC node)"

> If the block had been produced more than 5 epochs ago, try to send your request to an archival node

Let's add a reference on "archival node" to this section: https://docs.near.org/docs/develop/node/intro/types-of-node#archival-node (please, extend the section with URLs to mainnet and testnet archival nodes)

> The requested `account_id` has not been found while viewing

Maybe extend it with: "The requested `account_id` has not been found while viewing since the account has not been created or has been already deleted"

(apply the same about public keys and maybe some other places)

> Request a different block

"Specify a different block or retry if you request the latest state"

> The requested `contract_code` has not been found while viewing

"The account does not have any contract deployed on it"

There are no sane solutions to this one, I believe. Maybe suggest using a different account or deploying a contract to the queried account, but that is a bit of a stretch.

`NO_CONTRACT_CODE` should not happen on View contract state.

Please, double-check if `REQUEST_ROUTED` can actually happen on `broadcast_tx_commit`.

> Note that this does not mean transaction execution is failed not it means everything is fine.

"Note this does not mean the transaction execution failed nor it means everything is fine."

There is a typo in `error.cause.info.transactioc_hash`

`UNKNOWN_TRANSACTION` and `UNKNOWN_RECEIPT`:

> The requested transaction does not exist

"The requested transaction is not available on the node since it might have not been recorded on the chain yet or has been garbage-collected"

Solutions:
* Try again later
* If the transaction had been submitted more than 5 epochs ago, try to send your request to an archival node
* Check the transaction hash

File: docs/api/rpc/transactions.md [hunk 1]
```diff
@@ -176,6 +231,94 @@ http post https://rpc.testnet.near.org jsonrpc=2.0 id=dontcare method=broadcast_
...
+        <code>error.name</code>
+      </th>
+      <th>ERROR_CAUSE<br /><code>error.cause.name</code></th>
+      <th>Reason</th>
+      <th>Solution</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td rowspan="3">HANDLER_ERROR</td>
```
Comment (frol) [inline docs/api/rpc/transactions.md:276]:
```suggestion
      <td rowspan="2">HANDLER_ERROR</td>
```

image

File: docs/api/rpc/transactions.md [hunk 2]
```diff
@@ -176,6 +231,94 @@ http post https://rpc.testnet.near.org jsonrpc=2.0 id=dontcare method=broadcast_
...
+    "id": "dontcare",
+    "jsonrpc": "2.0"
+}
+```
+
+> **Heads up**
+>
+> The fields `code`, `data`, and `message` in the structure above are considered legacy ones and might be deprecated in the future. Please, don't rely on them.
+
+Here is the exhaustive list of the error variants that can be returned by `chunk` method:
```
Comment (frol) [inline docs/api/rpc/transactions.md:260]:
```suggestion
Here is the exhaustive list of the error variants that can be returned by `broadcast_tx_commit` method:
```

File: docs/api/rpc/transactions.md [hunk 3]
```diff
@@ -58,6 +58,61 @@ or [NEAR Explorer](https://explorer.testnet.near.org/) using the above `result`
...
```
Comment (frol) [inline docs/api/rpc/transactions.md:87]:
```suggestion
Here is the exhaustive list of the error variants that can be returned by `broadcast_tx_async` method:
```

Please, check the rest.

**Review by frol — APPROVED** [review]
I found a few minor bugs. Once those are resolved, I believe we are good to go and address the rest of the minor issues (if any) in separate PRs.

**Comment by frol** [comment]
@khorolets there are more places where the changes I proposed need to be copied over

**Comment by khorolets** [comment]
> @khorolets there are more places where the changes I proposed need to be copied over / re-applied

OK, I guess I've done it this time

**Comment by khorolets** [comment]
> @khorolets there are more places where the changes I proposed need to be copied over / re-applied

OK, I guess I've done it this time

**Comment by khorolets** [comment]
@thisisjoshford could you assist with failing test-links, please? :)

**Comment by bucanero** [comment]
> @thisisjoshford could you assist with failing test-links, please? :)

Hello @khorolets, I have reviewed/fixed some broken links, but there's still one dead URL that I don't know how to handle: `https://ropsten.bridgetonear.org/`

The server seems down so I don't know if it's a temporary issue or not. Maybe Josh knows better! 😄

**Comment by frol** [comment]
The failing test is not related to the PR, so I merged this PR. @bucanero @thisisjoshford Please, double check if we even need that bridge documentation with @djsatok or @artob

**Pull request merged**